Cache/Main.lean

```diff
@@ -83,8 +92,22 @@ def main (args : List String) : IO Unit := do
   let goodCurl ← pure !curlArgs.contains (args.headD "") <||> validateCurl
   if leanTarArgs.contains (args.headD "") then validateLeanTar
   let get (args : List String) (force := false) (decompress := true) := do
-    let hashMap ← if args.isEmpty then pure hashMap else hashMemo.filterByFilePaths (toPaths args)
-    getFiles hashMap force force goodCurl decompress
+    let mut getArgs := toPaths args
+    if args.isEmpty then
+      -- Find all .lean files in subdirectories (excluding .lake) of the current directory.
+      let allFiles := System.FilePath.walkDir (← IO.Process.getCurrentDir)
+        (fun p ↦ pure (p.fileName != some ".lake"))
+      let leanFiles := (← allFiles).filter (fun p ↦ p.extension == some "lean")
+      -- For each file, find all imports starting with Mathlib.
+      let mut allModules := #[]
```

Comment:
How about make this a `NameSet`? You're going to generate a lot of duplicates.

Reply:
Good idea; done.